src/perl/lib/wtsi_clarity/mq/messages/flowcell/control.pm

```diff
@@ -0,0 +1,96 @@
+package wtsi_clarity::mq::messages::flowcell::control;
+
+use Moose;
+use MooseX::Storage;
+
+our $VERSION = '0.0';
+
+with Storage( 'traits' => ['OnlyWhenBuilt'],
+              'format' => 'JSON',
+              'io' => 'File' );
+
+my @defaults = ( is => 'ro', isa => 'Str', required => 1 );
+
+has [
+     'tag_sequence',
+     'tag_set_name',
+     'entity_type',
+     'sample_uuid',
+     'study_uuid',
+     'entity_id_lims',
+    ] => @defaults;
+
+has 'tag_index' => @defaults, isa => 'Int';
+
+1;
+
+__END__
+
+=head1 NAME
+
+wtsi_clarity::mq::messages::flowcell::control
+
+=head1 SYNOPSIS
+
+  my $control = wtsi_clarity::mq::messages::flowcell::control->new(
+    sample_uuid                => '00000000-0000-0000-00000003',
+    study_uuid                 => '00000000-0000-0000-00000004',
+    tag_index                  => 3,
+    entity_type                => 'library_indexed_spike',
+    tag_sequence               => 'ATAG',
+    tag_set_id_lims            => '2',
```

Comment:
Didn't actually need it in the end so it's not added to the message. Will remove from synopsis.